### PR TITLE
feat/pipeline

### PR DESCRIPTION
**Pull request by JarbasAl**
configurable intent stage

### CODE REVIEW

File: ovos_core/intent_services/__init__.py [hunk 1]
```diff
@@ -199,6 +223,38 @@ def disambiguate_lang(message):
 
         return default_lang
 
+    def get_pipeline(self, skip_converse=False, skip_fallback=False):
+        # Create matchers
+        # TODO - from plugins
```
Comment (JarbasAl) [inline ovos_core/intent_services/__init__.py:228]: will solve this TODO in https://github.com/OpenVoiceOS/ovos-core/pull/310

for now just want to make it configurable to clean up padatious/padacioso and allow more control for users

File: ovos_core/intent_services/__init__.py [hunk 2]
```diff
@@ -199,6 +222,38 @@ def disambiguate_lang(message):
 
         return default_lang
 
+    def get_pipeline(self, skip_converse=False, skip_fallback=False):
```
Comment (forslund) [inline ovos_core/intent_services/__init__.py:225]: Perhaps add a docstring with explaining what the returned pipeline is a list of matcher functions, either a default list or as configured with 
```json
[...]
  "intents": {
    "pipeline": ["xxx", "yyy", ...]
  }
[...]
```

File: ovos_core/intent_services/__init__.py [hunk 3]
```diff
@@ -392,24 +435,8 @@ def handle_get_intent(self, message):
...
-            self.adapt_service.match_intent,
-            # self.fallback.high_prio,
-            padatious_matcher.match_medium,
-            # self.fallback.medium_prio,
-            padatious_matcher.match_low,
-            # self.fallback.low_prio
-        ]
         # Loop through the matching functions until a match is found.
-        for match_func in match_funcs:
+        for match_func in self.get_pipeline(skip_converse=True, skip_fallback=True):
```
Comment (forslund) [inline ovos_core/intent_services/__init__.py:439]: should the skips be hard-coded or should it be a set/tuple/list of matchers to skip?

File: ovos_core/intent_services/converse_service.py [hunk 4]
```diff
@@ -48,6 +48,7 @@ def deactivate_skill(self, skill_id, source_skill=None):
                 self.active_skills.pop(idx)
                 self.bus.emit(
                     Message("intent.service.skills.deactivated",
+                            {"skill_id": skill_id},
```
Comment (forslund) [inline ovos_core/intent_services/converse_service.py:51]: I'm sure this is correct that this is duplicated to both data and context? but maybe to be clear use data=..., context=...
?
(same a bit lower...)

**Review by forslund — COMMENTED** [review]
A couple of minor nit-picks and things that confused me :)

File: ovos_core/intent_services/__init__.py (hunk 3 above)
Comment (JarbasAl) [reply ovos_core/intent_services/__init__.py:439]: these are the special cases that should be skipped when getting registered intents via bus api, just added these for backwards compat but this whole api needs some re thinking
- converse / fallback / common qa need to be accounted for
- need to be made to report if they will handle an intent, without doing so, which doesnt play nice with any existing skills

i think for now we are kinda stuck with hardcoding these 2 until they work better, hopefully no more stages willl be added to the pipeline that behave like these 2

File: ovos_core/intent_services/converse_service.py (hunk 4 above)
Comment (JarbasAl) [reply ovos_core/intent_services/converse_service.py:51]: in ovos we add skill_id to the context of all bus messages, these were missing in a couple places i noticed during hivemind testing (which meant LocalHive was not mapping these to the correct skill to route messages)

probably could go into a different PR, just had this branch checked out while working on it

File: requirements/requirements.txt [hunk 5]
```diff
@@ -6,13 +6,13 @@ combo-lock>=0.2.2, <0.3
 padacioso~=0.2, >=0.2.1a8
 adapt-parser>=1.0.0, <2.0.0
 
-ovos_bus_client<0.1.0, >=0.0.5
+ovos_bus_client<0.0.6a1, >=0.0.5
```
Comment (JarbasAl) [inline requirements/requirements.txt:9]: TODO https://github.com/OpenVoiceOS/ovos-bus-client/commit/6eb7c564980333019bf9b2545a5c51b5d824fe1c introduced some issue where unittests hang forever `test/unittests/skills/test_event_container.py ....                       [ 34%]`

**Pull request merged**